client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -180,6 +250,24 @@ else if (annotationType.equals(BodyParam.class)) {
                     }
                 }
             }
+
+            final Class<?> returnType = method.getReturnType();
+            final boolean isAsync = (returnType == Single.class || returnType == Completable.class);
+            methodProxyDetails.setIsAsync(isAsync);
+            if (!isAsync) {
+                methodProxyDetails.setReturnType(returnType);
+            }
+            else {
+                final String asyncMethodName = method.getName();
```

Comment:
This is needed because we need to know the return type for the asynchronous method's parameterized type (T in Single<T> or Observable<T>) for deserialization.